### PR TITLE
bump sentry-native/0.4.15

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
Specify library name and version:  **sentry-native/0.4.15**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
Need to add newer cmake version as build requirement.

```python
def build_requirements(self):
     self.build_requires("cmake/3.22.0")
```

**Comment by uilianries** [comment]
Need to update conanfile.py on test package too:

```python
def build_requirements(self):
        if self.settings.os == "Macos" and self.settings.arch == "armv8":
            # Workaround for CMake bug with error message:
            # Attempting to use @rpath without CMAKE_SHARED_LIBRARY_RUNTIME_C_FLAG being
            # set. This could be because you are using a Mac OS X version less than 10.5
            # or because CMake's platform configuration is corrupt.
            # FIXME: Remove once CMake on macOS/M1 CI runners is upgraded.
            self.build_requires("cmake/3.22.0")
```

**Review by SSE4 — DISMISSED** [review]

**Comment by SSE4** [comment]
> Need to update conanfile.py on test package too:

okay, I hope very soon this workaround won't be needed

**Review by uilianries — DISMISSED** [review]

**Review by MartinDelille — CHANGES_REQUESTED** [review]
Since you removed support for the 0.2.6 version, you could also remove the patch files specific to this version:
- sentry-native/all/patches/0.2.6-0001-remove-sentry-handler-dependency.patch
- sentry-native/all/patches/0.2.6-0002-set-cmake-cxx-standard-14.patch

**Comment by AndreyMlashkin** [comment]
@icarusdes , approving something in conancenter is always a long process unfortunately.
In best case it takes a week

**Review by SSE4 — DISMISSED** [review]

**Review by uilianries — DISMISSED** [review]

**Comment by AndreyMlashkin** [comment]
@jgsogo  @SpaceIm @prince-chrismc @danimtb
please approve

File: recipes/sentry-native/all/conanfile.py [hunk 1]
```diff
@@ -130,6 +130,7 @@ def validate(self):
             raise ConanInvalidConfiguration("This version doesn't support ARM compilation")
 
     def build_requirements(self):
+        self.build_requires("cmake/3.22.0")
```
Comment (SpaceIm) [inline recipes/sentry-native/all/conanfile.py:133]:
```suggestion
        if tools.Version(self.version) >= "0.4.0" and self.settings.os == "Windows":
            self.build_requires("cmake/3.22.0")
```

Comment (SpaceIm) [reply recipes/sentry-native/all/conanfile.py:133]: Introduced in 0.4.0 and only for Windows: https://github.com/getsentry/sentry-native/blob/0.4.0/CMakeLists.txt#L1-L8

File: recipes/sentry-native/all/conandata.yml [hunk 2]
```diff
@@ -8,45 +11,11 @@ sources:
...
-    sha256: "35afb62eecc5e719187c81f5c89b4e3d3c45c7b3c1597836202989fa40c79178"
-  "0.4.7":
-    url: "https://github.com/getsentry/sentry-native/releases/download/0.4.7/sentry-native.zip"
-    sha256: "fb16658b250c342ed05e4d2edeff8ec806d9db758dfa188b78070bdfaf54f598"
-  "0.4.1":
-    url: "https://github.com/getsentry/sentry-native/releases/download/0.4.1/sentry-native.zip"
-    sha256: "ffeec2a8aa6d2f274123cd16c62ad6d5d4c6f993e44e9e2f88210261ccb51fcc"
-  "0.2.6":
-    url: "https://github.com/getsentry/sentry-native/releases/download/0.2.6/sentry-native-0.2.6.zip"
-    sha256: "0d93bd77f70a64f3681d4928dfca6b327374218a84d33ee31489114d8e4716c0"
```
Comment (SpaceIm) [inline recipes/sentry-native/all/conandata.yml:28]: Is it a good idea? Shouldn't we keep at least one version of each minor version?

Comment (AndreyMlashkin) [reply recipes/sentry-native/all/conandata.yml:28]: I am not really sure, if this is still supported by sentry server

File: recipes/sentry-native/all/conanfile.py [hunk 3]
```diff
@@ -130,6 +130,8 @@ def validate(self):
             raise ConanInvalidConfiguration("This version doesn't support ARM compilation")
 
     def build_requirements(self):
+        if tools.Version(self.version) >= "0.4.0" and self.settings.os == "Windows":
```
Comment (MartinDelille) [inline recipes/sentry-native/all/conanfile.py:133]: Currently all version are upper to 0.4.0 so we might remove this:

```suggestion
        if self.settings.os == "Windows":
```

**Review by MartinDelille — CHANGES_REQUESTED** [review]

**Comment by MartinDelille** [comment]
@AndreyMlashkin I don't have opinion on keepking 0.2.6 but maybe putting it back would make everyone happy and we'll get this version bump finally merged! 😄

**Comment by MartinDelille** [comment]
I don't know if the CI error is due to related or not to the 0.2.6 version but if it is it would be a good reason to drop support for it. Anyone has insight about what happened?

**Comment by AndreyMlashkin** [comment]
@jgsogo @SpaceIm @prince-chrismc @danimtb @SSE4 @uilianries 
please approve

**Review by MartinDelille — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Comment by SSE4** [comment]
> I don't know if the CI error is due to related or not to the 0.2.6 version but if it is it would be a good reason to drop support for it. Anyone has insight about what happened?

I think it wasn't related. all the errors I can see for this PR were random network errors.

**Review by SpaceIm — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

**Comment by Dev-vlad** [comment]
> Need to add newer cmake version as build requirement.
> 
> ```python
> def build_requirements(self):
>     self.build_requires("cmake/3.22.0")
> ```
 @SpaceIm @danimtb 

I don't think that CI build issue of conan center should be solved in the package recipe. This change forces all users that have a newer CMake version installed to force build it from sources, which takes much time.

**Pull request merged**